### PR TITLE
BAU: cleaning code

### PR DESCRIPTION
**Pull request by gokyo**

### CODE REVIEW

File: app/controllers/PdfDownloadController.scala [hunk 1]
```diff
@@ -39,47 +40,49 @@ class PdfDownloadController @Inject()(appConfig: FrontendAppConfig,
...
-        case Some(eori) => getApplicationPDF(eori, reference)
-        case None => Future.successful(Redirect(controllers.routes.SessionExpiredController.onPageLoad()))
-      }
-    } else {
-      Future.successful(Redirect(controllers.routes.UnauthorisedController.onPageLoad()))
-    }
+    getPdf(reference, token, getApplicationPDF)
   }
 
   def ruling(reference: String, token: Option[String]): Action[AnyContent] = identify.async { implicit request =>
```
Comment (gokyo) [inline app/controllers/PdfDownloadController.scala]: there were two methods doing exactly the same thing... so I created a generic method used by both

File: app/controllers/PdfDownloadController.scala [hunk 2]
```diff
@@ -39,47 +40,49 @@ class PdfDownloadController @Inject()(appConfig: FrontendAppConfig,
...
+  private def getRulingPDF(implicit request: Request[AnyContent]): (String, String) => Future[Result] = { (eori, reference) =>
+    caseService.getCaseWithRulingForUser(eori, reference) flatMap { c: Case =>
+      generatePdf(
+        rulingPdf(appConfig, c, c.decision.getOrElse(throw new IllegalStateException("Missing decision"))), s"BTIRuling$reference.pdf"
+      )
     }
   }
 
-  private def getApplicationPDF(eori: String, reference: String)(implicit r: Request[_]): Future[Result] = {
+  private def getApplicationPDF(implicit request: Request[AnyContent]): (String, String) => Future[Result] = { (eori, reference) =>
```
Comment (ed0906) [inline app/controllers/PdfDownloadController.scala]: I think the signature of this was better before. Its not very intuitive now

File: app/controllers/PdfDownloadController.scala [hunk 3]
```diff
@@ -39,47 +40,49 @@ class PdfDownloadController @Inject()(appConfig: FrontendAppConfig,
...
+        case Some(eori) => pdf(eori, reference)
         case None => Future.successful(Redirect(controllers.routes.SessionExpiredController.onPageLoad()))
       }
-    } else {
-      Future.successful(Redirect(controllers.routes.UnauthorisedController.onPageLoad()))
+      case _ => Future.successful(Redirect(controllers.routes.UnauthorisedController.onPageLoad()))
+    }
+  }
+
+  private def getRulingPDF(implicit request: Request[AnyContent]): (String, String) => Future[Result] = { (eori, reference) =>
```
Comment (ed0906) [inline app/controllers/PdfDownloadController.scala]: I think the signature of this was better before. Its not very intuitive now

File: app/controllers/PdfDownloadController.scala (hunk 2 above)
Comment (gokyo) [reply app/controllers/PdfDownloadController.scala]: I know, but I need a function if I want to remove repetitions from the code

Comment (gokyo) [reply app/controllers/PdfDownloadController.scala]: what do u suggest?

Comment (ed0906) [reply app/controllers/PdfDownloadController.scala]: It is a function, you can just change the way it is called.

Comment (gokyo) [reply app/controllers/PdfDownloadController.scala]: How do you like it?

**Comment by gokyo** [comment]
@ed0906 
Do you like it more now?

**Pull request merged**